worker-message-prioritization-distribution/src/main/java/com/github/workerframework/workermessageprioritization/redistribution/shovel/ShovelDistributor.java

```diff
@@ -122,34 +129,14 @@ public ShovelDistributor(
     }
     
     public void run() throws InterruptedException {
-        try {
-            while(true) {
-                runOnce();
-
-                try {
-                    Thread.sleep(distributorRunIntervalMilliseconds);
-                } catch (final InterruptedException e) {
-                    LOGGER.warn("Interrupted {}", e.getMessage());
-                    throw e;
-                }
-            }
-        } finally {
-            try {
-                nonRunningShovelCheckerExecutorService.shutdownNow();
-            } catch (final Exception exception) {
-                LOGGER.warn("Failed to shutdown nonRunningShovelCheckerExecutorService", exception);
-            }
-
-            try {
-                nonRunningShovelCheckerExecutorService.shutdownNow();
```

Comment:
This was the original reason the app was not shutting down, nonRunningShovelCheckerExecutorService should have been shovelRunningTooLongCheckerExecutorService here, meaning the shovelRunningTooLongCheckerExecutorService was not getting shut down.

We don't need to call `shutdownNow` though as we're using `MoreExecutors.getExitingScheduledExecutorService`.